### PR TITLE
My courses page

### LINKED ISSUE #65: My courses page
We want to add a page where the student can add curses to his list and then he could assign grades and other manipulations on those courses.
The page should contain the list of the users courses and also a form to add new courses

### PR DESCRIPTION
**Pull request by or313**
This PR includes the UI and logic of the view of my courses page as well as a form and model modifications to add new courses and remove from the list.
Fix #65

### CODE REVIEW

**Comment by or313** [comment]
Test CI failes because there are no tests in the project please ignore that

File: grades/models.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
 from django.db import models
 from django.conf import settings
-# Each user can add, modify or remove course from is list
-# Each course contains coures id(default by Django), name, credinitials and grade
+from django.core.validators import MaxValueValidator, MinValueValidator
+
 
 
 class Course (models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE,
```
Comment (simzacks) [inline grades/models.py:8]: I believe this table is the list of courses given, and the grades table combines the user and choose for a grade.
If so, this table probably shouldn't have a user relation

Comment (Omeramsc) [reply grades/models.py:8]: +1

File: grades/templates/course_list.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% extends 'layouts/default/page.html' %} {% load bootstrap4 %} 
+{% load render_table from django_tables2 %} {% block content %}
+<h2>Your Courses</h2>
+
+<h5>Add a new course here:</h5>
```
Comment (Omeramsc) [inline grades/templates/course_list.html:5]: I'm not 100% following. 
shouldn't the user choose a course from a list of available courses and assign it to him with the details relevant to him (grade, etc)

**Comment by or313** [comment]
I think that wxplaining the reason for thos page will answer both of your questions.
The idea is for a user to login, add courses to hia liat and then he can apply different funxtions on them withing the platform like:
Attack a bookmark for a cours or add a grade to a course.
We realized that first the user need to add the courses and that the platform does not supply them
@simzacks @Omeramsc

**Comment by Omeramsc** [comment]
@or313 This means every student will have his own course instance, including its name and credentials, meaning different students can create the same course (leading to multiple instances of the same course), and manual work. 
I'm not sure that's the best way to do what you aim for. a better way would be to have a list of courses (or an admin user who can create set courses so the students will be able to use them), and then a connection/class for the connection of each student to each course.

**Comment by or313** [comment]
@Omeramsc I get what you are saying but thats a different approach to the product spec.
We designed a platform which is a tool for an individual student. A student adds his own bookmarks, courses and grades as this user will login into his account and its his alone.
So yeah some courses will be written twice but for different users, by design.

**Comment by Omeramsc** [comment]
If It's by design, then OK.
LGTM.

**Review by Omeramsc — APPROVED** [review]

**Pull request merged**